### DISCUSSION TITLE
Going from ToonSquid to Moho Pro

### DISCUSSION THREAD

**Discussion by DWrightAnimation** [keiwando/toonsquid#341] [General]
There was a Humble Bundle going on that allowed me to get Moho Pro 12.5, which allowed me to test out some of its features. After having tested it out, here are some things that make ToonSquid better than Moho, and some things that it should borrow from Moho in order to enhance its abilities.

**What ToonSquid Does Better Than Moho**
- The easing presets. Moho seems to do an ease in, or an ease out when it comes to certain easing presets, like bouncing. ToonSquid has that, but also has the ability to do both at the same time. 
- Transform Hierarchy. Moho does have this, but it's more tedious, where you have to make a bone layer, put all the drawing layers inside it, then modify it to work properly. 
- Frame by Frame Animation. Moho has this, but requires a specific layer to do so, and it doesn't have raster levels, so doing rough animation is a pain. ToonSquid has it at Toon Boom Harmony's level. 

**Things ToonSquid Should Borrow From Moho**
- Being able to change control points on Vector lines made with the brush tool. ToonSquid's vector brushes are great, but they lack the ability to be transformed. The pen tool is fine, but it doesn't have the ability to have different textures or different sizes to emulate pressure. 
- Smart Bones. Obviously Moho's biggest selling point, being able to apply it to any bone and be able to alter the way the drawing reacts to the bone movement, such as altering how the arm reacts to the bone moving. 
- Blob Tool. This one basically lets you draw the inside of a shape like a brush stroke, but creates the stroke outside of the shape when you're done. 
- 3D Objects. Somehow, Moho let's you make 3D objects, possibly for things like props. Not sure if it can import 3D models from Blender or Nomad Sculpt, but imagine if ToonSquid could do that.

**Features That Could Be Useful**
- Being able to export to any other animation software. Imagine if you could export a ToonSquid project into applications such as Toon Boom, TVPaint, Moho, Toonz, Animate, etc, or be able to export as a Photoshop and/or Illustrator document, since Moho can take those kind of documents.

**Comment by Cubishon** [comment]
Having more advanced rigging tools would make ToonSquid ideal animation software since it is iPad app and both Moho or TOon Boom require desktop/laptop with drawing monitor setup.

**Comment by keiwando** [comment]
Thanks for the detailed insights!

**Comment by Cat-Butler** [comment]
Just some random and possibly disjointed thoughts:

The bones in Moho may take a little longer to rig than the ToonSquid transform hierarchy, but a huge advantage is that it allows for inverse kinematics when animating (drag the hand to re-pose the entire arm, or even get the figure to lean over if required).
Would love to see some sort of inverse kinematics in ToonSquid.

I wonder if another custom rotate tool might almost be enough to give a more bones-like forward kinematics experience with the existing transform hierarchy - instead of showing the normal transform box when a layer is selected, show a rotate-only tool starting at the layer pivot point and ending with a user-adjustable handle point relative to the pivot point (would need to switch between editing the handle location and moving the handle to rotate the layer).

As I understand it, ToonSquid's vector brushes aren't currently vector lines/curves, somewhat more like a pixel/raster brush run through a threshold and some sort of auto-trace (or at least that's how I think of them).
What would be more equivalent would be a freehand pencil-style tool (potentially with brushes, although I'd be happy with variable width control on a solid line) for Path layers (although I'm sure it will increase the calls to merge Path layers - at least conceptually, if not behind the scenes).
And, of course, @DWrightAnimation's earlier suggestion of keyframing control points on a Path object from frame markers in some sort of Path-based symbol.
